x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/TokenListCategorizer.java

```diff
@@ -206,7 +206,7 @@ public TokenListCategory mergeWireCategory(SerializableTokenListCategory seriali
         return mergedCategory;
     }
 
-    private synchronized TokenListCategory computeCategory(
+    private synchronized TokenListCategory computeCategory( // NOCOMMIT why synchronized? in ESQL at least there are no threads around
```

Comment:
I must have done this because `categoriesByNumMatches` gets both iterated and modified in this method. If it was being called in two different threads then that could result in one of the calls returning a crazy category, and/or corruption to the list. If it’s not possible for two threads to be using the same categorizer object then I guess it’s redundant and can be removed.

Reply:
In ESQL we absolutely won't use two threads. Aggs generally don't either, but we've made some changes to them recently that let them be more thready. I'll have to look.